lib/msal-common/src/unifiedCache/UnifiedCacheManager.ts

```diff
@@ -26,33 +38,71 @@ export class UnifiedCacheManager {
     }
 
     /**
-     * setter  for in cache memory
+     * sets the inMemory cache
+     * @param cache
      */
     setCacheInMemory(cache: InMemoryCache): void {
         this.inMemoryCache = cache;
     }
 
     /**
-     * get the cache in memory
+     * get the inMemory Cache
      */
     getCacheInMemory(): InMemoryCache {
         return this.inMemoryCache;
     }
 
     /**
      * Initialize in memory cache from an exisiting cache vault
+     * @param cache
      */
     generateInMemoryCache(cache: string): InMemoryCache {
-        return Deserializer.deserializeAllCache(Deserializer.deserializeJSONBlob(cache));
+        return Deserializer.deserializeAllCache(
+            Deserializer.deserializeJSONBlob(cache)
+        );
     }
 
     /**
      * retrieves the final JSON
+     * @param inMemoryCache
      */
     generateJsonCache(inMemoryCache: InMemoryCache): JsonCache {
         return Serializer.serializeAllCache(inMemoryCache);
     }
 
+    /**
+     * append credential cache to in memory cach
+     * @param accessToken
+     * @param idToken
+     * @param refreshToken
+     */
+    addCredentialCache(
+        accessToken: AccessTokenEntity,
+        idToken: IdTokenEntity,
+        refreshToken: RefreshTokenEntity
+    ): void {
+        this.inMemoryCache.accessTokens[
+            accessToken.generateCredentialKey()
+        ] = accessToken;
+        this.inMemoryCache.idTokens[
+            idToken.generateCredentialKey()
+        ] = idToken;
+        this.inMemoryCache.refreshTokens[
+            refreshToken.generateCredentialKey()
+        ] = refreshToken;
+    }
+
+    /**
+     * append account to the in memory cache
+     * @param account
+     */
+    addAccountEntity(account: AccountEntity): void {
+        const accKey = account.generateAccountKey();
+        if (!this.inMemoryCache.accounts[accKey]) {
```

Comment:
What if the existing `account` for this key is different than the passed in `account`? Or does that not matter?

Reply:
I think we overwrite it, the account key is based on homeAccountId so there is little chance of this being overwritten with an incorrect account.

Reply:
I have added overWrite as "remove" + "save" to be closer to a DB implementation.